### ISSUE TITLE
Maximum CSV file size for uploading data files is too small

### ISSUE THREAD

**Issue by SRGDamia1** [ODM2/ODM2DataSharingPortal#364]
A file with ~2months of data for 7 variables is too large.  That is much too small a limit.
[SL082_2018-05-18_2018-07-12_forUpload.zip](https://github.com/ODM2/ODM2DataSharingPortal/files/2750358/SL082_2018-05-18_2018-07-12_forUpload.zip)

**Comment by dbressler75** [comment]
Ensign and I were able to upload larger files using methods in the manual (https://www.envirodiy.org/mayfly-sensor-station-manual/#programming-and-activating-an-envirodiy-sensor-station); however, CTD temp did not work ("failed to add new sensor") and others too - see issue new issue (Some parameters cannot be added when uploading data)

**Comment by aufdenkampe** [comment]
@dbressler75, the manual link you pointed to refers to uploading an Arduino .ino sketch from your computer to a Mayfly. Is that what you are talking about?

This issue that @SRGDamia1 posted refers to uploading a CSV file with data to the data portal. Have you had success with that?

**Comment by dbressler75** [comment]
See image below of the manual upload section in manual that we used.  Yes it is related to Sara's issue.  We were able to upload CTD cond and depth data but not temp.   

![image](https://user-images.githubusercontent.com/39309124/55834092-1c48f780-5ae7-11e9-9c3a-a92c5297bb7c.png)

**Comment by SRGDamia1** [comment]
@aufdenkampe Is there an ETA on this fix?

**Comment by aufdenkampe** [comment]
From @dbressler75 via email on July 9:

> So we've been testing the upload function and found that it can accept up to 5000 rows of data (i.e., about two weeks worth of data points, collected every five minutes).  The issue we're having is that we have files that have a lot more data than that, so in order to upload all of these we would need to chunk the files down to 5000 then upload individually (see attached).  We can prioritize sites and we have an intern that can work on this but ideally MonMW could accept much larger files.  So before we started on this we wanted to see if you/Limnotech was planning to update this feature anytime soon?  
>  
> Sara was considering writing some code to automate the process as well.  We'll adjust as needed if no plans are in place to update MonMW in the near term, but we just wanted to be sure before we proceeded.

**Comment by aufdenkampe** [comment]
Closing this because it is a duplicate of #346, which also has more detailed information on potential causes and solutions.

Please move comments and conversation to #346.

**Comment by aufdenkampe** [comment]
@SRGDamia1 & @dbressler75,

There is a possibility that this issue has been resolved with some of the other CSV-related bug fixes for R0.10, or some of the recent server configuration changes that we have made in the last two months.

Please re-test on our staging servers at https://staging.monitormywatershed.org​.

Also please share the site ID and CSV files that you used for testing, so that we can try to repeat the error. 

Some of the issues described above (i.e. working for some but not all variables) appear to be syntax issues, some of which are described in #12 and summarized in https://github.com/ODM2/ODM2DataSharingPortal/issues/294#issuecomment-429695024.

In the duplicate issue #346 (Gateway timeouts when uploading CSV files), @SRGDamia1 saw a server timeout after 1 minute, and @jcaraballo17 suggested that perhaps the web server was infinitely looping and killed the request.

If there is no looping, but it just takes a long time, we might be able to do a simple fix by just increasing the time before the Gateway times out (actually, I don't know how simple that is or not).

If we need to speed up the upload and ingestion, that might require us to Use Pandas for CSV upload and download functions (#312), which is a bigger effort that will need to wait until Release 0.12

**Comment by SRGDamia1** [comment]
There seems to be a hard file-size limit of ~1MB - any files larger than that have the POST immediately fail with 413/Request entity too large.  With 5 parameters, each with a small number of significant figures, at 5 minute intervals, that's less than four months of data - ie, *definitely not enough.*  I think the file size limit is django induced and would have to changed in the settings.

When I tried uploading a file that was just a tiny bit less than 1MB, I got a gateway timeout in the developer console and a "Failed" pop-up, but several minutes later received an email saying the data had been uploaded and, indeed, new data up to the end of the file appeared in the sparklines.

**Comment by aufdenkampe** [comment]
@SRGDamia1, thanks for that testing and related info. It's very helpful.

**Comment by jcaraballo17** [comment]
Incresed the file size limit to 100MB since nginx caps it at 1MB by default.

**Comment by aufdenkampe** [comment]
@SRGDamia1, please retest on staging with your large file.

**Comment by SRGDamia1** [comment]
Now I can upload the file, but the request times out and the data isn't appended.

**Comment by SRGDamia1** [comment]
@aufdenkampe and @jcaraballo17 I emailed you some files.

**Comment by aufdenkampe** [comment]
@jcaraballo17, it looks like there is still an issue with this.

**Comment by aufdenkampe** [comment]
@SRGDamia1 and @jcaraballo17, is there a portion of this issue that isn't fixed? 
If so, let's refactor the issue to focus on what's remaining, and put that new issue in the next milestone.

**Comment by SRGDamia1** [comment]
Yes, while it will not accept a larger file, it will still time out and fail to upload.  I haven't tested the exact limit, but it's less than a year worth of 5 minute data, which is too small.

**Comment by SRGDamia1** [comment]
And, yes, I think stations should be visited and files downloaded [much] more frequently than that, but it doesn't always happen.

**Comment by SRGDamia1** [comment]
It's back to only accepting ~5000 lines at once.  That's a ridiculously small amount.

Also, any blank lines in the file (like extra new lines at the end) will cause the upload to fail.

**Comment by aufdenkampe** [comment]
@htaolimno and @tslawecki, it seems that Juan implemented an NGINX hot fix for this issue (i.e increasing file size limit) on Aug. 1 (see https://github.com/ODM2/ODM2DataSharingPortal/issues/364#issuecomment-517454944), but it got reverted on production when we issued Release 0.10 on Aug. 8.

Can you find and fix this NGINX setting on both production and staging, via our private config repo at https://github.com/LimnoTech/ODM2DataSharingPortalConfig/tree/master/nginx? 
Could you also update our documentation at https://github.com/LimnoTech/ODM2DataSharingPortalConfig/wiki/Config-File-Git-Tracking-&-SymLinks

**Comment by htaolimno** [comment]
@SRGDamia1 @aufdenkampe I checked the nginx's "client_max_body_size" setting. Both of them are "100M" which is what Juan had set. Can you confirm that you still have the same file size issue ?

**Comment by aufdenkampe** [comment]
@htaolimno, thanks for checking that Nginx isn't the issue. I think we'll need to bump this fix to the next milestone ([Release 0.12 - Tech Debt / Refactor Code](https://github.com/ODM2/ODM2DataSharingPortal/milestone/14)), in which we'll redo the CSV file handling to use the Pandas library.

**Comment by mbarneytu** [comment]
I have a station with 9 variables, and can only upload about 3000 rows, or less than 300 KB. A 3500-row upload fails (315 KB).
[BearCT_shortened_3500rows.zip](https://github.com/ODM2/ODM2DataSharingPortal/files/4150298/BearCT_shortened_3500rows.zip)

**Comment by dbressler75** [comment]
RE @mbarneytu last post, we've been doing it in 1-week chunks (about 1500 rows) because even 10 days (3000 rows) won't go sometimes (if I'm remembering correctly).

**Comment by aufdenkampe** [comment]
@SRGDamia1 & @dbressler75, we believed we just fixed with with our Restore CSV Upload Function commit (47047b1bb0f376fdf30924fb44fdd904c604b2f0) that was part of:
- #526 

Closing, but please feel free to test further.

We'll continue to work on CSV uploads and in the future when we fully implement:
- #312

**Comment by aufdenkampe** [comment]
@dbressler75, we believe that we addressed this issue with our v0.12.0 release, which went to production in early December, 2021.

Are you still seeing an issue?

**Comment by dbressler75** [comment]
@aufdenkampe, I haven't checked it in awhile so if you fixed it that's great.  Thank you!
